nemo/collections/asr/models/label_models.py

```diff
@@ -319,97 +318,56 @@ def training_step(self, batch, batch_idx):
 
         return {'loss': loss}
 
-    def validation_step(self, batch, batch_idx, dataloader_idx: int = 0):
+    def validation_step(self, batch, batch_idx, dataloader_idx: int = 0, tag: str = 'val'):
         audio_signal, audio_signal_len, labels, _ = batch
         logits, _ = self.forward(input_signal=audio_signal, input_signal_length=audio_signal_len)
         loss_value = self.loss(logits=logits, labels=labels)
         acc_top_k = self._accuracy(logits=logits, labels=labels)
         correct_counts, total_counts = self._accuracy.correct_counts_k, self._accuracy.total_counts_k
         self._macro_accuracy.update(preds=logits, target=labels)
-        self._auroc.update(preds=logits, target=labels)
+        stats = self._macro_accuracy._get_final_stats()
 
         return {
-            'val_loss': loss_value,
-            'val_correct_counts': correct_counts,
-            'val_total_counts': total_counts,
-            'val_acc_top_k': acc_top_k,
+            f'{tag}_loss': loss_value,
+            f'{tag}_correct_counts': correct_counts,
+            f'{tag}_total_counts': total_counts,
+            f'{tag}_acc_micro_top_k': acc_top_k,
+            f'{tag}_acc_macro_stats': stats
         }
 
-    def multi_validation_epoch_end(self, outputs, dataloader_idx: int = 0):
-        val_loss_mean = torch.stack([x['val_loss'] for x in outputs]).mean()
-        correct_counts = torch.stack([x['val_correct_counts'] for x in outputs]).sum(axis=0)
-        total_counts = torch.stack([x['val_total_counts'] for x in outputs]).sum(axis=0)
+    def multi_validation_epoch_end(self, outputs, dataloader_idx: int = 0, tag: str = 'val'):
+        loss_mean = torch.stack([x[f'{tag}_loss'] for x in outputs]).mean()
+        correct_counts = torch.stack([x[f'{tag}_correct_counts'] for x in outputs]).sum(axis=0)
+        total_counts = torch.stack([x[f'{tag}_total_counts'] for x in outputs]).sum(axis=0)
 
         self._accuracy.correct_counts_k = correct_counts
         self._accuracy.total_counts_k = total_counts
         topk_scores = self._accuracy.compute()
+        
+        self._macro_accuracy.tp = torch.stack([x[f'{tag}_acc_stats'][0] for x in outputs]).sum(axis=0)
+        self._macro_accuracy.fp = torch.stack([x[f'{tag}_acc_stats'][1] for x in outputs]).sum(axis=0)
+        self._macro_accuracy.tn = torch.stack([x[f'{tag}_acc_stats'][2] for x in outputs]).sum(axis=0)
+        self._macro_accuracy.fn = torch.stack([x[f'{tag}_acc_stats'][3] for x in outputs]).sum(axis=0)
```

Comment:
is this val_acc_stats or val_acc_macro_stats?

Reply:
Thanks for pointing out, I've fixed that~